### PR TITLE
Add Horizontal- and VerticalForm and SubmitButton. This allows submit…

### PR DESCRIPTION
**Pull request by mkotsbak**
…ting forms using enter key with onSubmit on the form

### CODE REVIEW

File: js/src/main/scala/org/widok/bindings/Bootstrap/package.scala [hunk 1]
```diff
@@ -141,6 +148,8 @@ package object Bootstrap {
     def element = DOM.createElement("button", contents)
   }
 
+  def SubmitButton(contents: View*) = Button(contents: _*).attribute("type", "submit")
```
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/Bootstrap/package.scala]: This is not related to Bootstrap. You could add a helper function to [``Button``](https://github.com/widok/widok/blob/master/js/src/main/scala/org/widok/bindings/HTML.scala#L81) to set the ``type`` more easily.

Comment (mkotsbak) [inline js/src/main/scala/org/widok/bindings/Bootstrap/package.scala]: You are right. But we should probably have Boostrap buttons too, with classes "btn btn-default"?

Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/Bootstrap/package.scala]: We already have ``Button`` in ``Bootstrap``, which inherits from ``ButtonBase`` and thus has the classes ``btn btn-default``. We could probably define ``ButtonBase`` in ``HTML``, so that ``type`` is available on HTML and Bootstrap buttons.

Comment (mkotsbak) [inline js/src/main/scala/org/widok/bindings/Bootstrap/package.scala]: Maybe time for more refactoring there. Added some in the PR.

File: js/src/main/scala/org/widok/bindings/HTML.scala [hunk 2]
```diff
@@ -79,8 +79,11 @@ object HTML {
 
   case class Button(contents: View*) extends Widget[Button] {
     val rendered = DOM.createElement("button", contents)
+    def buttonType(value: String) = attribute("type", value)
   }
 
+  def SubmitButton(contents: View*) = Button(contents: _*).buttonType("submit")
```
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: I don't think we need this function.

File: js/src/main/scala/org/widok/bindings/HTML.scala [hunk 3]
```diff
@@ -79,8 +79,11 @@ object HTML {
 
   case class Button(contents: View*) extends Widget[Button] {
     val rendered = DOM.createElement("button", contents)
+    def buttonType(value: String) = attribute("type", value)
```
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: What do you think about:

```scala
def `type`(value: String) = attribute("type", value)
```

File: js/src/main/scala/org/widok/bindings/HTML.scala (hunk 2 above)
Comment (mkotsbak) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: Ok, you mean let users type Button(something).buttonType("submit") ?

In that case I don't like "submit" string. It should be replaced with an Enum like type.

File: js/src/main/scala/org/widok/bindings/Bootstrap/package.scala [hunk 4]
```diff
@@ -123,9 +130,9 @@ package object Bootstrap {
   }
 
   trait ButtonBase[T] extends Widget[T] { self: T =>
```
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/Bootstrap/package.scala]: You could declare ``ButtonBase`` in ``HTML`` and extend from it here. Then you don't need to set the underlying element.

```scala
trait ButtonBase[T] extends HTML.ButtonBase[T] { self: T =>
  ...
}
````

File: js/src/main/scala/org/widok/bindings/HTML.scala (hunk 2 above)
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: I agree. You can create a ``sealed trait`` hierarchy with all possible button types.

File: js/src/main/scala/org/widok/bindings/HTML.scala (hunk 3 above)
Comment (mkotsbak) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: Avoided it because it is a bit awkward to type, but usually the IDE helps you, so maybe better?

Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: Alternatively, you could also write ``tpe``, which is quite common.

**Comment by mkotsbak** [comment]
I've cleaned up the PR and incorporated all things we have discussed here I think.

File: js/src/main/scala/org/widok/bindings/HTML.scala [hunk 5]
```diff
@@ -77,8 +77,22 @@ object HTML {
...
-    val rendered = DOM.createElement("button", contents)
+  sealed trait ButtonType { val tpe: String }
+  object ButtonType {
+    case object Button extends ButtonType { val tpe = "button" }
+    case object Submit extends ButtonType { val tpe = "submit" }
+    case object Reset extends ButtonType { val tpe = "reset" }
+  }
+
+  trait ButtonBase[T] extends Widget[T] { self: T =>
+    def element: dom.html.Element
```
Comment (tindzk) [inline js/src/main/scala/org/widok/bindings/HTML.scala]: You can delete ``element`` and ``rendered``.

**Comment by tindzk** [comment]
Great, thanks!
